client/client.go

```diff
@@ -1134,6 +1131,23 @@ func (c *Client) updateNodeFromDriver(name string, info *structs.DriverInfo) *st
 	return c.configCopy.Node
 }
 
+// updateNodeFromFingerprint updates the node with the result of
+// fingerprinting the node from the diff that was created
+func (c *Client) updateNodeFromDevices(devices []*structs.NodeDeviceResource) {
+	c.configLock.Lock()
+	defer c.configLock.Unlock()
+
+	// Not updating node.Resources: the field is deprecated and includes
+	// dispatched task resources and not appropriate for expressing
+	// node available device resources
+	if !structs.DevicesEquals(c.config.Node.NodeResources.Devices, devices) {
```

Comment:
I initially went the Fingerprint route, but then realized that all updates to resources happen independently by type (i.e. cpu vs memory vs disk), so going fingerprint route didn't provide much benefit.

So using `updateNodeFromDevices` directly as it's simpler.  Also confirmed that updates to node are accumulative properly (with a test), and fingerprint manager gets updated node and subsequent fingerprint updates don't unnecessarily update node or stomp values.  So I think this is good.